bdk_core/src/tx_graph.rs

```diff
@@ -182,11 +189,110 @@ impl TxGraph {
             }
         }
     }
+    
+    pub fn persist_graph<T: Write + Seek>(
+        &self,
+        writer: &mut T,
+        changeset: Vec<(&Txid, &TxNode)>,
+    ) -> io::Result<()> {
+        // TODO: write current state_id + version
+        let header_bytes = [0u8; 8];
+        let _size = writer.write(&header_bytes)?;
+        let _pos = writer.seek(SeekFrom::End(0))?;
+        let mut buf = Vec::new();
+        for (txid, tx_node) in changeset {
+            match tx_node {
+                TxNode::Whole(tx) => {
+                    let serialized_tx = encode::serialize(tx);
+                    // length of the transaction data
+                    buf.extend((serialized_tx.len() as u32).to_be_bytes().to_vec());
+                    // 1u8 after the length stands for the kind of transaction (Whole).
+                    buf.push(1u8);
+                    // add the transaction data
+                    buf.extend(serialized_tx);
+                }
+                TxNode::Partial(outpoints) => {
+                    buf.extend(((outpoints.len()) as u32).to_be_bytes().to_vec());
+                    buf.push(2u8);
+                    let serialized_txid = encode::serialize(txid);
+                    // add txid
+                    buf.extend(serialized_txid);
+                    for (idx, output) in outpoints {
+                        let serialized_txout = encode::serialize(output);
+                        let idx = (*idx as u32).to_be_bytes().to_vec();
+                        // add length of new item (idx + output)
+                        buf.extend(((idx.len() + serialized_txout.len()) as u32).to_be_bytes());
+                        // add index
+                        buf.extend(idx);
+                        // add output
+                        buf.extend(serialized_txout);
+                    }
+                }
+            }
+        }
+        let _size = writer.write(&buf)?;
+        Ok(())
+    }
+
+    pub fn from_db<T: Read + Seek>(
+        &mut self,
+        reader: &mut T,
+    ) -> Result<HashMap<Txid, TxNode>, Box<dyn Error>> {
```

Comment:
I think there's no need to return what was read here. If you want to see it you can just create an empty `TxGraph` and read it in, have a look and then extend to your existing `TxGraph`.

We should make a custom `Error` enum here.

Reply:
I will change the API here